vignettes/articles/installation-troubleshooting.Rmd

```diff
@@ -0,0 +1,71 @@
+---
+title: "Installation troubleshooting"
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+This page details common problems encountered when trying to install igraph. If you did not find an answer to your problem here, feel free to ask for help on https://igraph.discourse.group/ When you do so, be sure to state: 
+
+1. the output of `R.version` 
+1. where you obtained R from 
+1. the complete output from running `install.packages("igraph")` in a fresh R session.
+
+## Cannot compile igraph from sources on Windows or macOS
+
+Do not compile igraph from sources, unless you know what you are doing! It is much more convenient to use the binaries from CRAN instead. These can be installed using `install.packages("igraph")`.
+
+Usually, it takes a few days for binary releases to become available on CRAN after each new igraph source release. During this period, `install.packages()` will give you a choice about using the latest source release or the previous binary release. Just choose to use the binary release, i.e. respond "no" to "Do you want to install from sources?"
+
+```
+> install.packages('igraph')
+
+  There is a binary version available but the source
+  version is later:
+       binary source needs_compilation
+igraph  1.2.7  1.2.8              TRUE
+
+Do you want to install from sources the package which needs compilation? (Yes/no/cancel) no
+```
+
+CRAN provides Windows and macOS binaries only for the last two minor releases of R (e.g. 4.2 and 4.1), but not for older ones (e.g. 4.0). Make sure that you are using one of these supported R versions. Refer to https://r-project.org/ to find out what the latest R version is at the moment.
+
+Note that the pre-compiled binaries on CRAN are only compatible with the R distribution [provided by CRAN itself](https://www.r-project.org/). If you obtained R from different sources, such as Homebrew or MacPorts on macOS, you will not be able to use these, and the only way to install packages will be to compile them from sources.
+
+#### I still want to compile igraph from sources
+
+If you decide to try to compile from sources anyway, basic requirements are listed below. It is assumed that you are comfortable compiling software from sources and resolving basic issues as they arise. Otherwise, please use the binaries.
+
+ - Make sure you have a compatible toolchain installed, including a Fortran compiler. On Linux, make sure you have `gfortran` installed. For Windows, see https://cran.r-project.org/bin/windows/Rtools/
+ - Make sure you have the GLPK C library installed. This is not the same as the Rglpk R package. Instructions for installing libraries for Rtools4 on Windows are [here](https://cran.r-project.org/bin/windows/Rtools/), see in particular the "Using pacman" link. You will find a copyable pacman command for installing all dependencies [here](https://github.com/igraph/rigraph/#installation).
```

Comment:
Better links are (better because they contain text) now
- https://cran.r-project.org/bin/windows/Rtools/rtools40.html (until R4.1.3)
- https://cran.r-project.org/bin/windows/Rtools/rtools42/rtools.html

Reply:
@maelle I wrote the troubleshooting info originally. If you think that something should be changed, feel free to go ahead and do it, as you know much more about R than I do.

Reply:
ok, thanks! FWIW I think these links changed recently-ish.